client/main.lua

```diff
@@ -419,19 +419,19 @@ end
 ---@param shop string
 local function startTestDriveTimer(time, shop)
     local gameTimer = GetGameTimer()
+    local timeMs = tonumber(1000 * time)
+
     CreateThread(function()
         while InTestDrive do
-            if GetGameTimer() < gameTimer + tonumber(1000 * time) then
-                local secondsLeft = GetGameTimer() - gameTimer
-                if secondsLeft >= tonumber(1000 * time) - 20 then
+            local currentGameTime = GetGameTimer()
+            local secondsLeft = currentGameTime - gameTimer
+            if currentGameTime < gameTimer + timeMs then
+                if secondsLeft >= timeMs - 50 then
```

Comment:
I'm a bit confused by this logic. I would have just expected condition `if secondsLeft <= 0`

Reply:
I am going to be honest, I did not try to fix the logic, just modify the currently bad one and try to make sure the condition is met when very few seconds are left, let me give it a go and test that.

Reply:
Revisited this with a couple of prints, secondsLeft value goes up from 0 instead of the way around, so using the above condition immediately stops the test, so this does seem to be correct.